Core.ApplicationServices/SystemUsage/Migration/ItSystemUsageMigrationService.cs

```diff
@@ -36,87 +36,107 @@ public class ItSystemUsageMigrationService : IItSystemUsageMigrationService
+            _itSystemUsageRelationsService = itSystemUsageRelationsService;
             _domainEvents = domainEvents;
         }
 
-        public Result<IReadOnlyList<ItSystem>, OperationFailure> GetUnusedItSystemsByOrganization(
+        public Result<IQueryable<ItSystem>, OperationError> GetUnusedItSystemsByOrganizationQuery(
             int organizationId,
-            string nameContent,
             int numberOfItSystems,
-            bool getPublicFromOtherOrganizations)
+            bool getPublicFromOtherOrganizations,
+            params IDomainQuery<ItSystem>[] conditions)
         {
-            if (string.IsNullOrWhiteSpace(nameContent))
+            if (numberOfItSystems is < 1 or > 25)
             {
-                throw new ArgumentException(nameof(nameContent) + " must be string containing more than whitespaces");
-            }
-            if (numberOfItSystems < 1)
-            {
-                throw new ArgumentException(nameof(numberOfItSystems) + " Cannot be less than 1");
+                return new OperationError($"{nameof(numberOfItSystems)} must satisfy constraint: 1 <= n <= 25", OperationFailure.BadInput);
             }
 
             var dataAccessLevel = _authorizationContext.GetDataAccessLevel(organizationId);
 
             if (dataAccessLevel.CurrentOrganization < OrganizationDataReadAccessLevel.Public)
             {
-                return OperationFailure.Forbidden;
+                return new OperationError("User is not allowed to access organization", OperationFailure.Forbidden);
             }
 
+            var subQueries = new List<IDomainQuery<ItSystem>>();
+
+            subQueries.AddRange(conditions);
+
             var queryBreadth = getPublicFromOtherOrganizations ? OrganizationDataQueryBreadth.IncludePublicDataFromOtherOrganizations : OrganizationDataQueryBreadth.TargetOrganization;
-            var unusedSystems = _systemRepository.GetUnusedSystems(new OrganizationDataQueryParameters(organizationId, queryBreadth, dataAccessLevel));
+            var baseQuery = _systemRepository.GetUnusedSystems(new OrganizationDataQueryParameters(organizationId, queryBreadth, dataAccessLevel));
+
+            var result = subQueries.Any()
+                ? new IntersectionQuery<ItSystem>(subQueries).Apply(baseQuery)
+                : baseQuery;
 
-            //Refine, order and take the amount requested
-            var result = unusedSystems
-                .ByPartOfName(nameContent)
+            var finalQuery = result
                 .Where(x => x.Disabled == false)
                 .OrderBy(x => x.Name)
-                .Take(numberOfItSystems)
-                .ToList()
-                .AsReadOnly();
+                .Take(numberOfItSystems);
+
+            return Result<IQueryable<ItSystem>, OperationError>.Success(finalQuery);
+        }
 
-            return result;
+        public Result<IReadOnlyList<ItSystem>, OperationError> GetUnusedItSystemsByOrganizationByName(
```

Comment:
This method is going to be killed once the UI is updated to use V2

Reply:
that requires us to extend v2 search with another query param rights?